linalg/densemat.cpp

```diff
@@ -436,6 +455,43 @@ double DenseMatrix::Det() const
             d[3] * (d[2] * d[7] - d[1] * d[8]) +
             d[6] * (d[1] * d[5] - d[2] * d[4]);
       }
+      case 4:
+      {
+         const double *d = data;
+         return
+            d[ 0] * (d[ 5] * (d[10] * d[15] - d[11] * d[14]) -
```

Comment:
That's a good point.  I needed this for 4x4 matrices so I didn't put much thought into the recursive algorithm (although I did test it).

I'll implement a version using the LUFactors class and run some comparisons to find the crossover point.